milestones/api.py

```diff
@@ -268,25 +284,36 @@ def add_course_content_milestone(course_key, content_key, relationship, mileston
         course_key=course_key,
         content_key=content_key,
         relationship=relationship,
-        milestone=milestone)
+        milestone=milestone,
+        requirement=requirement
+    )
 
 
-def get_course_content_milestones(course_key, content_key, relationship=None):
-    """
-    Retrieves the set of milestones for a given course content module
-    'relationship': optional filter on milestone relationship type (string, eg: 'requires')
-    Returns an array of dicts containing milestones
+def get_course_content_milestones(course_key=None, content_key=None, relationship=None, user=None):
```

Comment:
Hmm, ok -- maybe it makes sense, then, to add a guard clause to the function to prevent the 'list all' use case, assuming we don't want to support it.